### ISSUE TITLE
✨ New `ZeroAsIntSerializer` type

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#695]
## 📝 Description

Similarly to #690, in the `types-kotlinx-serialization` subproject, we want to introduce the `ZeroAsIntSerializer` **experimental** type for serializing the `Zero` type as `Int`.

```kotlin
class ZeroAsIntSerializer : KSerializer<Zero> {
    override val descriptor: SerializerDescriptor

    override fun equals(other: Any?): Boolean
    override fun hashCode(): Int
    override fun serialize(encoder: Encoder, value: Zero)
    override fun deserialize(decoder: Decoder): Zero
    override fun toString(): String
}
```

This serializer should be available for all Kotlin platforms.

## 🔗 Dependencies

This issue is blocked by the following items:

- [x] #717 
- [x] #763 
- [x] #784

## ✅ Checklist

- [ ] ✨ Add the type and its public constructor with tests, documentation and samples.
- [ ] ✨ Override the `toString` function with tests, documentation and samples.
- [ ] ✨ Override structural equality operations (`equals` and `hashCode` functions) with tests, documentation and samples.
- [ ] ✨ Make the type implementing the `KSerializer<Zero>` type with tests, documentation and samples.
- [ ] 📝 Update the unreleased changelog for this issue.
- [ ] 📝 After closing this issue, attach tracking issues to a milestone.

**Comment by LVMVRQUXL** [comment]
Superseded by #793.
